api/envoy/extensions/filters/http/aws_request_signing/v3/aws_request_signing.proto

```diff
@@ -64,3 +64,11 @@ message AwsRequestSigning {
   // When applied, all headers that start with "x-envoy" and headers "foo" and "bar" will not be signed.
   repeated type.matcher.v3.StringMatcher match_excluded_headers = 5;
 }
+
+message AwsRequestSigningPerRoute {
+  // Override the global configuration of the filter with this new config.
```

Comment:
From the impl, this seems to override the whole message, so a clarification here would be helpful (i.e. this is not merging at field level in `AwsRequestSigning`).

Reply:
Wonder if merging at field level is better than override the entire message?
If we decide to merge at field level we will need to add each of the fields from AwsRequestSigning to AwsRequestSigningPerRoute and may need to maintain two messages here.

Thoughts?

Reply:
Not saying merging field level is better or not, it depends on the possible use cases. Overriding a whole message generally is less confusing, but if you expect most case just overriding one field (e.g. `hostname`), then a field level merging could avoid repetitive config in every route, especially when other fields like `match_excluded_headers` is large.

Reply:
Yeah, I think `match_excluded_headers` could get large and possible we just want to override on a few selected fields. 
Leaning towards modifying my changes to support field level override.

Reply:
Added clarification